src/medley/core.cljc

```diff
@@ -202,8 +224,14 @@
   that elements with the same key are overwritten, rather than added to a
   vector of values."
   {:added "1.2.0"}
-  [f coll]
-  (persistent! (reduce #(assoc! %1 (f %2) %2) (transient {}) coll)))
+  ([f]
```

Comment:
The transducer version doesn't make things distinct. Let's consider this as part of a separate PR.

Reply:
Hmm, I didn't consider that case - if you call `(into {})` on the result it would of course end up being distinct, having to check for equality would slow things down and might defeat the point of using transducers.
But you're right, I'll remove it from this PR too for now :)